src/modules/simulation/gz_bridge/CMakeLists.txt

```diff
@@ -64,6 +64,8 @@ if(gz-transport_FOUND)
 			GZMixingInterfaceESC.hpp
 			GZMixingInterfaceServo.cpp
 			GZMixingInterfaceServo.hpp
+			GZMixingInterfaceMotor.cpp
+			GZMixingInterfaceMotor.hpp
```

Comment:
The GZMixingInterfaceESC is supposed to model the motor, which controls speed. Is there a reason you want a separate interface here?

Reply:
Other than the reverse, it is doing exactly the same no? If we create a new interface for every vehicle type, this will quickly become out of hands

Reply:
Yes other than reverse it is the same, except the encoder section. But I believe that with these 3 we cover all cases, we have the "drone" esc's that don't require reverse, the servos (position control) and now motors (speed control + reverse).

Reply:
> we have the "drone" esc's that don't require reverse

Yes, but drone ESCs can also reverse and already used in the omnicopter. So it is better to make the ESC interface configurable and use it for all vehicles.

Reply:
Just a consideration: larger motor controllers used for wheels contain PID logic and ramp up/down timing parameters, as well as current limits and feedback/odometry. Classic Gazebo plugin allowed to simulate some of it (via <joint_control_pid> node). Scaling and offset were also useful there.

Reply:
I'll take a look and see what I can find @slgrobotics